packages/data-service/src/data-service.ts

```diff
@@ -2285,6 +2314,48 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return { name, ...normalized };
   }
 
+  @op(mongoLogId(1_001_000_063))
+  async previewUpdate(
+    ns: string,
+    filter: Document,
+    update: Document
+  ): Promise<UpdatePreview> {
+    const coll = this._collection(ns, 'CRUD');
+    const session = this._startSession('CRUD');
+    try {
+      session.startTransaction();
+
+      const docsToPreview = await coll
+        .find(filter, { session })
+        .sort({ _id: 1 })
```

Comment:
Don’t we want to $sample?

Reply:
That's a good point. With the current design, $sample will return different documents each time this function is called. For consistency, it will be more obvious to the user if we use always the same documents for the preview.

I would use $sample if the ids are decided before calling this function, as we can store them in the store for example, to reuse them, but it's not the case here. However, suggestions accepted!

Reply:
I think we decided against adding explicit UI elements for re-sampling, but I _do_ think there should be some way for getting a fresh (set of) sample document(s) when the current ones aren’t not helpful (e.g. because the update does not affect it).

Practically speaking, that would mean closing and re-opening the modal then. How that actually translates into the DataService interface is a different question then, of course :slightly_smiling_face: But given that it probably does make sense to get the _ids of the documents we want to look up here inside the transaction, I’d maybe lean towards also doing the sampling here.

Reply:
An easy way to resample might be to just take an optional `skip` parameter. Then we could just add pagination to the UI. Then the sample is still consistent between calls AND the user can pick different docs to preview. We can probably add that once we need it, though.

For what its worth: You'd need a reasonably complicated update for it to not affect every doc that matches the filter, right? I mean a simple $set should affect every doc one way or another.